src/lxc/lxccontainer.c

```diff
@@ -2320,7 +2320,16 @@ static bool remove_from_array(char ***names, char *cname, int size)
 {
 	char **result = get_from_array(names, cname, size);
 	if (result != NULL) {
-		free(result);
+		size_t i = result - *names;
+		free(*result);
+		memmove(*names+i, *names+i+1, (size-i-1) * sizeof(char*));
+		char **newnames = (char**)realloc(*names, (size-1) * sizeof(char *));
+		if (!newnames) {
+			ERROR("Out of memory");
+			return false;
```

Comment:
The `size` here is tracked by the *caller*, and we just erased an element (`memmoved` a range of elements over it).
The above will log an error, but we still return `false` to the caller, indicating that the element was *not* actually removed and will assume the array has not changed in length.
This means that the caller will now see the final element twice.
I think it's better to just return `true` in either case. Having an extra element allocated and not used is harmless. Having an element in there twice on the other hand may lead to double free corruptions later down the line.

Reply:
Indeed, well spotted, thanks